File: cap03/greet.py
'''
Objetivo: explicar a ideia de pilha de chamada.

Pilha de chamada é uma estrutura de dados que guarda as variáveis
de múltiplas funções

'''


def greet(name):
    print("Hello, " + name + "!")
    greet2(name)
    print("Working to say bye...")
    bye()

def greet2(name):
    print("How are " + name + "?")

def bye():
    print("Ok, bye!")


greet('Igor')

File: cap03/greet.js
// call stack is a very important concept
// we know that call stack has two operations: push and pop
// one important thing about call stack is that can get really big and occupies a lot of memory

function greet(name) {
  console.log(`Hi, ${name}!`)
  greet2(name)
  console.log('Working to say bye...')
  bye()
}

function greet2(name) {
  console.log(`How are you, ${name}?`)
}

function bye() {
  console.log('Ok! Bye!')
}

greet('Igor')
